src/Chapter1.hs

```diff
@@ -653,7 +670,11 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
+firstDigit :: Int -> Int
+firstDigit n
+  | (abs n) < 10 = (abs n)
+  | otherwise = firstDigit(div (abs n) 10) 
```

Comment:
That is right 💯 

You can see that `abs n` is reused several times in this function. This usually means that you can move it into the `where` case and reuse everywhere 👍🏼 

Also note, that these parentheses are redundant and can be removed safely 🙂 
In Haskell, you always separate functions and arguments with spaces even if argument has parenthesis around 👾 

```suggestion
  | abs n < 10 = abs n
  | otherwise = firstDigit (div (abs n) 10) 
```

Reply:
Yes, you are right. And while writing this solution I realised it as well, but I wasn't sure how to fix it. I'm gonna try and rewrite the solution moving abs into the where case.

Reply:
Okay, I found a way to use abs only once, but I don't think this is what you meant 😂  I just couldn't think of a way of how to move the use of abs into the where case, that didn't make the solution more complicated and longer... Would you maybe review this solution as well and give me some feedback, please? Because this also feels like it's not much of an improvement compared to my original solution.

`firstDigit :: Int -> Int
firstDigit n = getFirst (abs n)
  where 
    getFirst :: Int -> Int
    getFirst x 
      | x < 10 = x
      | otherwise = firstDigit(div x 10) `